### PR TITLE
Expose EASYRSA_PASSIN and EASYRSA_PASSOUT env variables

### LINKED ISSUE #365: allow external EASYRSA_PASSIN and EASYRSA_PASSOUT env vars
Hi @luizluca ,

thank you for your awesome PR #242 , it makes a lot of things easier. However I'm wondering, why did you forbid the external source for the `EASYRSA_PASSIN` and `EASYRSA_PASSOUT` variables? See https://github.com/OpenVPN/easy-rsa/blob/master/easyrsa3/easyrsa#L2303

I think allowing those env vars could provide additional flexibility, for example in situations, when the control of command line options is difficult.

Thanks for considering this request.

peter

### PR DESCRIPTION
**Pull request by pschiffe**
By exposing these variables it's possible to configure the password from
various sources by specifing env vars. This is a followup to PR #242

Fixes #365

### CODE REVIEW

**Comment by pschiffe** [comment]
cc @luizluca

**Comment by pschiffe** [comment]
The CI check failed with error `An error occurred while generating the build script.` - not sure what that means, all tests are passing - `Completed Sat 21 Mar 2020 06:58:02 PM CET (Total errors: 0)`. Could someone try to rerun that travis test?

**Comment by ecrist** [comment]
Looks like Travis might be having a problem with their Mac build boxes.

**Comment by luizluca** [comment]
The patch has many unrelated whitespace changes

**Comment by pschiffe** [comment]
Dropped the whitespace changes from the patch.

File: easyrsa3/vars.example [hunk 1]
```diff
@@ -220,3 +220,8 @@ fi
 
 #set_var EASYRSA_BATCH		""
 
+# These two variables allow the user to specify a source for password,
+# using any openssl password options like pass:1234 or env:var
+
+#set_var EASYRSA_PASSIN		""
```
Comment (luizluca) [inline easyrsa3/vars.example]: I don't see why someone would set this by default. passin/passout have different meanings for different commands. I would remove those lines.

**Comment by luizluca** [comment]
A malicious agent could set EASYRSA_PASSOUT and skip asking the admin for a password. If the admin inattentive does not notice that, a private key might be saved using the password defined by the malicious agent. Another attack would be to use EASYRSA_PASSOUT to destroy data overwriting a target file writable by easyrsa caller. Although, setting arbitrary env vars would be enough to potentially do much more harm than this case.

Another non-intentional problem would be someone leaving EASYRSA_PASSIN/OUT defined by mistake (used in a previous easyrsa call). It could make the command fail without a clear warning or save a private key with a previously used password. Maybe we should add a warning if data came from an env var.

Setting any random data to EASYRSA_PASSIN/OUT will only make command fail as its content is protected by double quotes and it always prefix it with '--passin/out' option.

I don't think that this patch will make too much harm and it does add an alternative way to pass arguments. Besides the vars.example change I suggested, it should be accepted.

File: easyrsa3/vars.example (hunk 1 above)
Comment (pschiffe) [reply easyrsa3/vars.example]: I meant this only for a reference / documentation. It's true that these variables are empty / not set by default, what is shown in that file.

The command `build-ca` requires both env vars to be set, and it makes sense to set it to the same `env:var` value.

Anyway, I just wanted to document it as best I could, but if you still think that this is not appropriate I can remove it. Please confirm. Thanks.

**Comment by ecrist** [comment]
That information is better documented in the EasyRSA-Advanced.md file.  I *do* think a mention in the config file specifying it is NOT recommended to be set there would be best.  Maybe even a check in easyrsa itself to refuse to use a config file that contains EASYRSA_PASSIN/OUT value?

File: easyrsa3/vars.example (hunk 1 above)
Comment (pschiffe) [reply easyrsa3/vars.example]: I've dropped the examples from config file.

**Comment by pschiffe** [comment]
@ecrist up to you - let me know what would you like to see here and I can try to do it. I didn't find any examples in the script how to warn that variable is set in the config file and I'm not sure if it's worth the effort and complexity, but if you think it's important..

**Comment by ecrist** [comment]
@pschiffe there's nothing in there now - a simple grep test for the config file would suffice.  I can help if needed.

**Comment by pschiffe** [comment]
I've included a commit with grep check for the env vars in the config file.

**Comment by ecrist** [comment]
I'll merge this, looks good on the surface.  Could use an update to unit tests in https://github.com/OpenVPN/easyrsa-unit-tests/issues/26

**Comment by Deantwo** [comment]
I cannot find any usage examples or documentation on how to use this at all.
Can the documentation be updated and maybe a link to it be posted here?

**Comment by luizluca** [comment]
You mean --passin/--passout arg? It is a standard openssl argument https://www.openssl.org/docs/manmaster/man1/openssl-passphrase-options.html

**Pull request merged**